### PR TITLE
fix: timeline not showing when set to public

### LINKED ISSUE #37855: Timeline does not appear for public portfolio
**Describe the bug**
Timeline can be marked as public in the account settings, but it does not appear for public viewers.

**To Reproduce**
1. Create an account
1. Complete at least one challenge
1. Go to [settings](https://www.freecodecamp.org/settings) and change the "My profile" and the "My timeline" settings to public
1. Click on "Show me my public portfolio"
1. Copy the URL and open in a new incognito window 

**Expected behavior**
I expected to see the timeline in both the logged-in user's browser and in the public view (the incognito window)

**Screenshots**

image

logged-in user:
image

incognito window:
image

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome & Firefox
 - Version: latest stable

### PR DESCRIPTION
**Pull request by moT01**
So the problem here was that, when requesting another users profile (not your own), the `completedChallenges` array was only sent back if `showCerts` and `showTimeLine` were `true` - aka public. So I filtered out the cert challenges if `showCerts` was private and sent it back like that. I came up with a few iterations of this logic before landing on what's in the PR. Feel free to suggest something better.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #37855

### CODE REVIEW

File: api-server/common/models/user.js [hunk 1]
```diff
@@ -796,7 +796,16 @@ export default function(User) {
...
-      completedChallenges: showCerts && showTimeLine ? completedChallenges : [],
+      completedChallenges: (function() {
+        if (showCerts && showTimeLine) {
+          return completedChallenges;
+        } else if (showTimeLine) {
+          return completedChallenges.filter(
+            challenge => challenge.challengeType !== 7
+          );
+        }
+        return [];
```
Comment (RandellDawson) [inline api-server/common/models/user.js]:
```suggestion
      completedChallenges: (function() {
        return showTimeLine
          ? showCerts
            ? completedChallenges
            : completedChallenges.filter(({ challengeType }) => challengeType !== 7)
         : [];
```

File: api-server/common/models/user.js [hunk 2]
```diff
@@ -796,7 +796,16 @@ export default function(User) {
       ...user,
       about: showAbout ? about : '',
       calendar: showHeatMap ? calendar : {},
-      completedChallenges: showCerts && showTimeLine ? completedChallenges : [],
+      completedChallenges: (function() {
```
Comment (RandellDawson) [inline api-server/common/models/user.js]:
```suggestion
      completedChallenges: (function() {
        return showTimeLine
          ? showCerts
            ? completedChallenges
            : completedChallenges.filter(({ challengeType }) => challengeType !== 7)
         : [];
```

Comment (RandellDawson) [reply api-server/common/models/user.js]: I tried to make a multi-line comment, but it did not look like it was going to work correctly if you committed it.  I can try again and can see what it looks like on your end.

Comment (moT01) [reply api-server/common/models/user.js]: Yea, I tried a similar nested ternary - but prettier was complaining about it and wouldn't let me commit it.

Comment (RandellDawson) [reply api-server/common/models/user.js]: @ojeytonwilliams Why would this be?

Comment (ojeytonwilliams) [reply api-server/common/models/user.js]: It's an eslint rule: [no-nested-ternary](https://eslint.org/docs/rules/no-nested-ternary).  

I'd go with

```js
if (showTimeLine) {
  return showCerts
    ? completedChallenges
    : completedChallenges.filter(
        challenge => challenge.challengeType !== 7
      );
} else {
  return [];
}
```

which it accepts and I think is fairly readable.

**Review by ojeytonwilliams — COMMENTED** [review]
I noticed, while reviewing this, at the certs never appear https://github.com/freeCodeCamp/freeCodeCamp/issues/37943, so the logic might need to change if we decide that that's intended behaviour.

However, this fixes the problem in #37855 so, when Randell's okay with the eslint stuff, I'm happy for this to be merged.

File: api-server/common/models/user.js (hunk 2 above)
Comment (RandellDawson) [reply api-server/common/models/user.js]: At least destructure the `challengeType` from the challenge object in the filter callback function.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**